update Ilab on OCP documentation (#294)
opendatahub-io/ilab-on-ocp: README.md

# InstructLab on Red Hat OpenShift AI

This repo is the central location for the code, Container files and yamls needed to deploy [Instructlab] onto an [OpenShift] cluster with [Red Hat OpenShift AI (RHOAI)]. This project leverages a number of the tools included with RHOAI working together to run InstructLab. Specifically, Data Science Pipelines for application orchestration, Kserve Serving for model serving, and the Distributed Training Operator to run our model training across multiple GPU enabled nodes.

<p align="center"><img src="assets/images/completed_pipeline.png" width=50%\></p>

## Getting Started

This project makes running the InstructLab large language model (LLM) fine-tuning process easy and flexible on OpenShift. However, before getting started there are a few prerequisites and additional setup steps that need to be completed.

## Requirements

* An OpenShift cluster with:

>   * Sufficient GPUs available for training.

Reviewer: I still think this suggestion can be added. We are testing the pipeline with just 1 NVIDIA L40S in Fede's cluster

See in this [blog post](https://blog.instructlab.ai/2024/12/end-to-end-testing-of-the-instructlab-cli/) what the InstructLab team uses for their e2e testing

>   * Sufficient GPUs available for training.

Reviewer: It's strange, I still see this:

![image](https://github.com/user-attachments/assets/16ab14f4-5025-48d7-82e2-4169682f2411)

Approving anyway (it can be fixed after)

    * NOTE: Install NVIDIA GPU Operator 24.6
        * This is due to a [known issue with CUDA Driver] versions mismatch, you can only use this Operator version.
* A [StorageClass] that supports dynamic provisioning with [ReadWriteMany] access mode.
  * You can deploy your own using [nfs storage] for non production use cases
* Have the location for the [base model configured] (i.e. S3 or OCI)
* A locally installed [oc] command line tool to create and manage kubernetes resources.
* [Teacher and Judge models] being served with their access credentials stored as k8s secrets in `<data-science-project-name/namespace>`
* An OCI registry to push the output model to along with credentials with push access

#### Disconnected Cluster Requirements

Follow the [disconnected setup instructions] for setting up your disconnected environment.

## Enable the pipeline

The InstructLab pipeline is automatically provisioned and managed by the DataSciencePipelinesApplication (DSPA) resource. This management is disabled by default, you may enable it by patching the following field in the DSPA:

```bash
DS_PROJECT="<data-science-project-name/namespace>"
DSPA_NAME="dspa" # RHOAI Default name for DSPA

> oc patch dspa ${DSPA_NAME} -n ${DS_PROJECT} --type=merge --patch='{"spec": { "apiServer": { "managedPipelines": { "instructLab": {"state": "Managed"}}}}}'

Reviewer: ```suggestion
oc patch dspa ${DSPA_NAME} -n ${DS_PROJECT} --type=merge --patch='{"spec": { "apiServer": { "managedPipelines": { "instructLab": {"state": "Managed"}}}}}'
After a few seconds, the InstructLab pipeline will be automatically added to the pipeline server and it will be available in RHOAI Dashboard.
```

```

## Output OCI Registry

> In your DS project namespace, deploy your OCI output registry secret:

Reviewer: ```suggestion
As a result of running the InstructLab pipeline, a fine-tuned model will be generated. The pipeline can push the resulting model to a OCI container registry (e.g. quay.io) if the credentials are provided. To do it, in your data science project namespace, deploy your OCI output registry secret:
```

```yaml
# oci_output_push_secret.yaml
apiVersion: v1
kind: Secret
metadata:
  name: <oci-registry-push-secret>
stringData:
  .dockerconfigjson: {...}
type: kubernetes.io/dockerconfigjson
```

Deploy it in your DS project:

```bash
oc -n <data-science-project-name/namespace> oc apply -f oci_output_push_secret.yaml
```

Note the `metadata.name` of this secret, you will need this when filling out the InstructLab pipeline parameters.

## Run the Pipeline

Now that all the cluster requirements have been set up, we are ready to upload and run our InstructLab pipeline!

Starting the InstructLab pipeline run is the same as starting any other run using Data Science Pipelines.

The instructions for how to create a typical run can be found [here][DSP Run Docs]. For the pipeline selection step, you will be able to select the `InstructLab` pipeline from the pipeline list when creating a run.

> You can find a description of each Parameter once you go to the run creation page. Some parameters already have defaults, while others do not, and some are optional. Please carefully read the description of each to better customize the pipeline to meet your needs.

Reviewer: ```suggestion
You can find a description of each Parameter once you go to the run creation page. Some parameters already have defaults, while others do not, and some are optional. Please carefully read the description of each to better customize the pipeline to meet your needs. 

Beware there is a [bug](https://issues.redhat.com/browse/RHOAIENG-22522) in RHOAI 2.19 where empty parameters do not appear when duplicating a run, so during your testing you might have to create a new pipeline runs instead of duplicating them.
```
